File: permit-system-be/internal/model/permit_license.go
package model

import "time"

type PermitLicense struct {
	BaseModel

	MasterDocumentID string `gorm:"index"`
	UploadedBy       string `gorm:"index"`
	UnitID           string `gorm:"index"`

	CurrentStatusID string `gorm:"index"`

	RelatedPrevDocumentID *string `gorm:"index"`

	DocumentName string
	Description  string

	FileURL  string
	FileSize int64

	ExpiredAt  time.Time `gorm:"index"`
	ApprovedAt *time.Time

	RejectedReason string

	IsActive bool `gorm:"default:true"`

	MasterDocument MasterDocument `gorm:"foreignKey:MasterDocumentID"`
	User           User           `gorm:"foreignKey:UploadedBy"`
	Unit           Unit           `gorm:"foreignKey:UnitID"`

	CurrentStatus ApprovalStatus `gorm:"foreignKey:CurrentStatusID"`

	RelatedPrevDocument *PermitLicense `gorm:"foreignKey:RelatedPrevDocumentID"`

	ApprovalHistories []ApprovalHistory `gorm:"foreignKey:PermitLicenseID"`
}

File: permit-system-fe/src/types/permit-license.ts
export interface PermitLicenseItem {
  id: string
  document_name: string
  description: string
  file_url: string
  file_size: number
  expired_at: string
  status: string
  uploaded_by: string
  unit: string

  master_document: {
    id: string
    name: string
  }

  related_prev_document_id?: string
}

export interface PaginationResponse<T> {
  page: number
  limit: number
  total_rows: number
  total_pages: number
  data: T[]
}

export interface CreatePermitLicensePayload {
  master_document_id: string
  document_name: string
  description: string
  expired_at: string

  related_prev_document_id?: string

  file: File
}